siege_utilities/integrations/parsons/_auth.py

```
@@ -0,0 +1,213 @@
+                service=cred.siege_service,
+                username=username,
+                field=cred.siege_field,
+            )
+        except Exception as exc:  # noqa: BLE001 — we translate at boundary
+            if cred.required:
+                raise ConnectorAuthError(
+                    f"Failed to resolve required credential "
+                    f"{cred.siege_service}/{cred.siege_field} "
+                    f"for connector {connector!r} "
+                    f"(profile={profile!r}): {exc}"
+                ) from exc
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)